engine/test/integration_tests/dm_many_tables/run.sh

```diff
@@ -19,22 +19,24 @@ function run() {
 	run_sql 'DROP DATABASE IF EXISTS dm_many_tables'
 	run_sql 'CREATE DATABASE dm_many_tables;'
 	for i in $(seq $TABLE_NUM); do
-		run_sql "CREATE TABLE dm_many_tables.t$i(i TINYINT, j INT UNIQUE KEY);"
+		run_sql --quiet "CREATE TABLE dm_many_tables.t$i(i TINYINT, j INT UNIQUE KEY);"
 		for j in $(seq 2); do
-			run_sql "INSERT INTO dm_many_tables.t$i VALUES ($j,${j}000$j),($j,${j}001$j);"
+			run_sql --quiet "INSERT INTO dm_many_tables.t$i VALUES ($j,${j}000$j),($j,${j}001$j);"
 		done
 		# to make the tables have odd number of lines before 'ALTER TABLE' command, for check_sync_diff to work correctly
-		run_sql "INSERT INTO dm_many_tables.t$i VALUES (9, 90009);"
+		run_sql --quiet "INSERT INTO dm_many_tables.t$i VALUES (9, 90009);"
 	done
 
 	# create job & wait for job finished
 	job_id=$(create_job "DM" "$CUR_DIR/conf/job.yaml" "dm_many_tables")
-	exec_with_retry --count 200 "curl \"http://127.0.0.1:10245/api/v1/jobs/$job_id/status\" | tee /dev/stderr | jq -e '.TaskStatus.\"mysql-01\".Status.Stage == \"Finished\"'"
+	# check progress is forwarded gradually, not jump to "finished"
+	exec_with_retry --count 200 "curl \"http://127.0.0.1:10245/api/v1/jobs/$job_id/status\" | tee /dev/stderr | jq -e '.task_status.\"mysql-01\".status.status | .finishedBytes > 0 and .finishedBytes < .totalBytes'"
+	exec_with_retry "curl \"http://127.0.0.1:10245/api/v1/jobs/$job_id\" | tee /dev/stderr | jq -e '.state == \"Finished\"'"
```

Comment:
L33 needs the job doesn't finish, L34 checks job is finished?

Reply:
yes, "check progress is forwarded gradually, not jump to finished"